src/Microsoft.AspNet.OData.Shared/PatchMethodHandler.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.OData
+{    
+    /// <summary>
+    /// Basic Interface for PatchMethodHAndler
+    /// </summary>
+    public interface IPatchMethodHandler
+    {
+
+    }
+
+    /// <summary>
+    /// Handler Class to handle users methods for create, delete and update
+    /// </summary>
+    public abstract class PatchMethodHandler<TStructuralType>: IPatchMethodHandler where TStructuralType : class
+    {
+        /// <summary>
+        /// TryCreate method to create a new object.
+        /// </summary>        
+        /// <param name="patchObject">The Delta<typeparamref name="TStructuralType"/> object to be patched over original object. Optional to patch</param>
+        /// <param name="createdObject">The created object (CLR or Typeless)</param>
```

Comment:
>Typeless) [](start = 67, length = 9)

this is only used for CLR-backed, right?

Reply:
Yes